__tests__/integration/integration.test.js

```diff
@@ -51,17 +48,114 @@ describe('POST /clients/signup', () => {
     expect(response.status).toBe(422);
     expect(response.body.error).toBe('Senhas diferentes.');
   });
+
   it('should return 409 when email already exists', async () => {
     const body = {
       name: 'test',
       email: 'test@test.com',
       password: '123456',
       confirmPassword: '123456',
     };
-    await db.query('INSERT INTO users (name, email, password) values ($1, $2, $3)', [body.name, body.email, body.password]);
+    await db.query('INSERT INTO users (name, email, password, type) values ($1, $2, $3, $4)', [body.name, body.email, body.password, 'CLIENT']);
+    
     const response = await agent.post('/clients/signup').send(body);
 
     expect(response.status).toBe(409);
     expect(response.body.error).toBe('Conflito de dados.');
   });
 });
+
+describe('POST /clients/signin', () => {
+  it('should return 200 when passed valid login data', async () => {
+
+    const bodyLogin = {
+      email: 'test@test.com',
+      password: '123456',
+    };
+
+    const response = await agent.post('/clients/signin').send(bodyLogin);
+    
+    expect(response.status).toBe(200);
+  });
```

Comment:
Este teste depende de outro, dado que só o teste da linha 27 insere esse usuário no banco. Modifique o `beforeAll` no início do arquivo para `beforeEach` e faça as alterações necessárias para que os testes sejam independentes.

Reply:
O primeiro teste do describe abaixo faz isso da forma correta.